data-service/src/main/java/com/amazonaws/blox/dataservice/handler/EnvironmentHandler.java

```diff
@@ -30,26 +31,71 @@
 
   @NonNull private EnvironmentRepository environmentRepository;
 
-  public Environment createEnvironment(final String environmentName, final String accountId)
-      throws EnvironmentExistsException, StorageException {
-    return null;
+  public Environment createEnvironment(final Environment environment)
+      throws EnvironmentExistsException, ServiceException {
+    try {
+      return environmentRepository.createEnvironment(environment);
+    } catch (final StorageException e) {
+      throw new ServiceException(
+          String.format(
+              "Exception occurred when creating environment with id %s and version %s",
+              environment.getEnvironmentId(), environment.getEnvironmentVersion()),
+          e);
+    }
   }
 
-  public Environment describeEnvironment(final String environmentId) {
-    return null;
+  public Environment createEnvironmentTargetVersion(final Environment environment)
```

Comment:
nit: Is this doing a `create`, or an upsert? It looks like the latter; in which case the name might be better off as something that more closely corresponds to the semantics, like `setEnvironmentVersionAsTarget`, or something similar.

Reply:
what's an upsert?

Reply:
oh this will be a conditional fail for existing environments, will fix